classes/security/UserGroupDAO.inc.php

```diff
@@ -779,46 +776,79 @@ function deleteSettingsByLocale($locale) {
 	 */
 	function _getSearchSql($searchType, $search, $searchMatch, &$params) {
 		$searchTypeMap = [
-			IDENTITY_SETTING_GIVENNAME => 'usgs.setting_value',
-			IDENTITY_SETTING_FAMILYNAME => 'usfs.setting_value',
-			USER_FIELD_USERNAME => 'u.username',
-			USER_FIELD_EMAIL => 'u.email',
-			USER_FIELD_AFFILIATION => 'us.setting_value',
+			IDENTITY_SETTING_GIVENNAME => "EXISTS(
+				SELECT 0
+				FROM user_settings usgs
+				WHERE usgs.user_id = u.user_id
+					AND usgs.setting_name = 'givenName'
+					AND LOWER(usgs.setting_value) LIKE LOWER(?)
+			)",
+			IDENTITY_SETTING_FAMILYNAME => "EXISTS(
+				SELECT 0
+				FROM user_settings usgs
+				WHERE usgs.user_id = u.user_id
+					AND usgs.setting_name = 'familyName'
+					AND LOWER(usgs.setting_value) LIKE LOWER(?)
+			)",
+			USER_FIELD_USERNAME => 'LOWER(u.username) LIKE LOWER(?)',
+			USER_FIELD_EMAIL => 'LOWER(u.email) LIKE LOWER(?)',
+			USER_FIELD_AFFILIATION => "EXISTS(
+				SELECT 0
+				FROM user_settings usgs
+				WHERE usgs.user_id = u.user_id
+					AND usgs.setting_name = 'affiliation'
+					AND LOWER(usgs.setting_value) LIKE LOWER(?)
+			)",
 		];
 
 		$searchSql = '';
-
+		$search = trim($search);
 		if (!empty($search)) {
-
 			if (!isset($searchTypeMap[$searchType])) {
-				$str = $this->concat('COALESCE(usgs.setting_value,\'\')', 'COALESCE(usfs.setting_value,\'\')', 'u.email', 'COALESCE(us.setting_value,\'\')');
-				$concatFields = ' ( LOWER(' . $str . ') LIKE ? OR LOWER(cves.setting_value) LIKE ? ) ';
-
-				$search = strtolower($search);
-
-				$words = preg_split('{\s+}', $search);
-				$searchFieldMap = array();
-
-				foreach ($words as $word) {
-					$searchFieldMap[] = $concatFields;
-					$term = '%' . $word . '%';
-					array_push($params, $term, $term);
-				}
-
-				$searchSql .= ' AND (  ' . join(' AND ', $searchFieldMap) . '  ) ';
+				$terms = array_map(function ($term) {
+					return "%$term%";
+				}, PKPString::regexp_split('/\s+/', $search));
+				$filters = [];
+				// Search the user table
+				$filters[] = '(1 = 1' . str_repeat(' AND ' . $this->concat('LOWER(u.email)', 'LOWER(u.username)') . ' LIKE LOWER(?)', count($terms)) . ')';
+				array_push($params, ...$terms);
+
+				// Search the user settings
+				$filters[] = '
+					EXISTS(
+						SELECT 0
+						FROM user_settings us
+						WHERE
+							u.user_id = us.user_id
+							' . str_repeat(' AND LOWER(us.setting_value) LIKE LOWER(?)', count($terms)) . '
+					)';
+				array_push($params, ...$terms);
```

Comment:
Hmmm, I'll probably submit an update to move this into the line 813 and use a `GROUP_CONCAT`/`STRING_AGGR`, otherwise the search won't be very effective.

Reply:
Updated... The performance is still great in my use-case.